### PR TITLE
Correct Fix to make sure debugger dependencies are copied

### PR DESCRIPTION
**Pull request by pieandcakes**
Fix double compile and make sure that it is only run once.

@bobbrow This should fix the items in the email you sent me.

### CODE REVIEW

File: Extension/src/Support/prepublish.js [hunk 1]
```diff
@@ -11,15 +11,25 @@
...
-// Required for nightly builds. Nightly builds do not enable CPPTOOLS_DEV.
-console.log(">> node ./out/src/Support/copyDebuggerDependencies.js");
-cp.execSync("node ./out/src/Support/copyDebuggerDependencies.js", {stdio:[0, 1, 2]});
+// If the required debugger file doesn't exist, make sure it is copied.
+if (process.env.CPPTOOLS_DEV || !fs.existsSync('./debugAdapters/bin/cpptools.ad7Engine.json')) {
+    const outDir = './out/src/Support/'
+    const copyDebuggerDependenciesJSFile = path.join(outDir, 'copyDebuggerDependencies.js');
+    if (!fs.existsSync('./out/src/Support/copyDebuggerDependencies.js'))
+    {
+        console.log(">> tsc -p ./src/Support/copyDebuggerDependencies.ts");
```
Comment (WardenGnaw) [inline Extension/src/Support/prepublish.js]: log should be the same as the exec command.
```tsc ./src/Support/copyDebuggerDependencies.ts --outDir " + outDir```

**Review by WardenGnaw — CHANGES_REQUESTED** [review]

**Review by sean-mcmanus — APPROVED** [review]

**Review by WardenGnaw — APPROVED** [review]

**Comment by bobbrow** [comment]
This change breaks the VSIX build.  The vsce.cmd only runs the prepublish step, so you need to tsc compile the whole src folder, not just the copy script.

**Comment by bobbrow** [comment]
I'm working on a fix.

**Comment by pieandcakes** [comment]
I wonder why this wasn't an issue when I tested it yesterday. Thanks for fixing it. I submitted your PR

**Comment by WardenGnaw** [comment]
@pieandcakes I just realized, didn't you also have changes to the package.json for your PR that called tsc in there? @bobbrow fix works too.

**Comment by pieandcakes** [comment]
@WardenGnaw I don't know how it didn't make it in but Bob fixed it.

**Pull request merged**